source/boundary_composition/spherical_constant.cc

```diff
@@ -41,30 +42,42 @@ namespace aspect
                           const Point<dim> &/*position*/,
                           const unsigned int /*compositional_field*/) const
     {
-      // Verify that the GeometryModel is a SphericalShell, Chunk, or an
+      // Verify that the GeometryModel is a Sphere, SphericalShell, Chunk, or an
       // EllipsoidalChunk since only for geometries based on spherical shells
       // do we know which boundary indicators are used and what they mean
       const GeometryModel::Interface<dim> *geometry_model = &this->get_geometry_model();
       const std::string boundary_name = geometry_model->translate_id_to_symbol_name(boundary_indicator);
 
-      if (dynamic_cast<const GeometryModel::SphericalShell<dim>*>(geometry_model) != 0
-          || dynamic_cast<const GeometryModel::Chunk<dim>*>(geometry_model) != 0
-          || dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*>(geometry_model) != 0)
+      if (dynamic_cast<const GeometryModel::Sphere<dim>*>(geometry_model) != 0)
+        {
+          if (boundary_name == "surface")
+            return surface_composition;
```

Comment:
I think I would prefer if this returned "outer_composition", and that we document this in the definition of the input parameters